db/sqldb/audit_log.go

```diff
@@ -88,6 +88,70 @@ func (d *DB) CreateAuditLog(tx db.Tx, a *pacta.AuditLog) (pacta.AuditLogID, erro
 	return id, nil
 }
 
+func (d *DB) TransferAuditLogOwnership(tx db.Tx, sourceUserID, destUserID pacta.UserID, sourceOwnerID, destOwnerID pacta.OwnerID) (int, error) {
+	auditLogsUpdated := -1
+	err := d.RunOrContinueTransaction(tx, func(tx db.Tx) error {
+		countLogsReferencingSource := func() (int, error) {
+			row := d.queryRow(tx, `
+			SELECT 
+				COUNT(*)
+			FROM audit_log
+			WHERE 
+				actor_id = $1 OR
+				primary_target_id = $1 OR
+				secondary_target_id = $1 OR
+				actor_owner_id = $2 OR
+				primary_target_owner_id = $2 OR
+				secondary_target_owner_id = $2;`, sourceUserID, sourceOwnerID)
```

Comment:
I think you want to put a `... FOR UPDATE` at the end here (at least for the first invocation, not sure about the second), e.g. `SELECT ... FROM ... WHERE ... FOR UPDATE`, see [1] [2]

[1] https://www.postgresql.org/docs/current/explicit-locking.html#LOCKING-ROWS
[2] https://www.postgresql.org/docs/current/sql-select.html#SQL-FOR-UPDATE-SHARE

Reply:
Mooted now that we don't do an update.